backend/hitas/tests/apis/test_api_owner.py

```diff
@@ -224,6 +293,53 @@ def test__api__owner__update__invalid_identifier_to_invalid(api_client: HitasAPI
     }
 
 
+@pytest.mark.django_db
+def test__api__owner__update__invalid_identifier_to_self(api_client: HitasAPIClient):
+    owner: Owner = OwnerFactory.create(identifier="foo")
+    data = {
+        "name": "Matti Meikäläinen",
+        "identifier": owner.identifier,
+        "email": "test@example.com",
+    }
+
+    url = reverse("hitas:owner-detail", kwargs={"uuid": owner.uuid.hex})
+    response = api_client.put(url, data=data, format="json")
+    assert response.status_code == status.HTTP_200_OK, response.json()
+    assert {
+        "id": owner.uuid.hex,
+        "name": data["name"],
+        "identifier": data["identifier"],
+        "email": data["email"],
+    } == response.json()
+
+
+@pytest.mark.django_db
+def test__api__owner__update__invalid_identifier_to_existing_invalid(api_client: HitasAPIClient):
+    owner_1: Owner = OwnerFactory.create(identifier="foo")
+    owner_2: Owner = OwnerFactory.create(identifier="bar")
+    data = {
+        "name": "Matti Meikäläinen",
+        "identifier": owner_2.identifier,
+        "email": "test@example.com",
+    }
+
+    url = reverse("hitas:owner-detail", kwargs={"uuid": owner_1.uuid.hex})
+    response = api_client.put(url, data=data, format="json")
+    assert response.status_code == status.HTTP_400_BAD_REQUEST, response.json()
```

Comment:
This should not result in an error, since the identifier is not valid.
e.g. There may be cases where we dont have the full identifier of a person, only the birthday, and we should be able to save multiple people with the same birthday to the system 😄
Also now it's not possible to have multiple owners with `identifier = None`

Reply:
This case is testing the update, so it's asserting that you shouldn't be able to change an existing owner's invalid identifier to some other owner's identifier, essentially merging them. Maybe that should be still allowed since no actual merging is happening. 

As for the creating part, I can make it not validate when adding new owners with invalid identifiers.

Reply:
Changes made, new tests also added for null identifiers.

Reply:
>Maybe that should be still allowed since no actual merging is happening?

I would say yes.

If it's not allowed, and multiple `Owner`'s have the same `identifier`, and one of them has only their name changed, a _'duplicate identifier error'_ is raised even though the identifier was unchanged

Reply:
`if value != self.instance.identifier` bypasses the checks if the identifier remains unchanged, so you could still change just the name and be fine. But I think allowing invalid identifiers to change to other invalid identifiers is fine based on what you said previously, so I'll make the merge.